tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,351 @@
+            case JAVA_SERIALIZABLE:
+                return new SerializableEmployee(id);
+            case IDENTIFIED_DATA_SERIALIZABLE:
+                return new IdentifiedDataSerializableEmployee(id);
+            case PORTABLE:
+                return new PortableEmployee(id);
+        }
+        return null;
+    }
+
+    //Custom Predicates
+    private static class NamePredicate implements Predicate<String, EmployeeInterface> {
+        private final String name;
+
+        private NamePredicate(String name) {
+            this.name = name;
+        }
+
+        @Override
+        public boolean apply(Map.Entry<String, EmployeeInterface> entry) {
+            EmployeeInterface value = entry.getValue();
+            return value.getName().equals(name);
+        }
+    }
+
+    public interface EmployeeInterface {
+
+        public String getName();
+    }
+
+    public static class DataSerializableEmployee implements DataSerializable,EmployeeInterface {
+
+        protected int id;
+        protected String name;
+
+        public String getName() {
+            return name;
+        }
+
+        public DataSerializableEmployee() {
+        }
+
+        public DataSerializableEmployee(int id) {
+            this.id = id;
+            name = names[random.nextInt(names.length)];
+        }
+
+        @Override
+        public void writeData(ObjectDataOutput objectDataOutput) throws IOException {
+            objectDataOutput.writeInt(id);
+            objectDataOutput.writeUTF(name);
+        }
+
+        @Override
+        public void readData(ObjectDataInput objectDataInput) throws IOException {
+            id = objectDataInput.readInt();
+            name = objectDataInput.readUTF();
+        }
+
+        @Override
+        public String toString() {
+            return "EmployeeInterface{" +
+                    "id=" + id +
+                    ", name='" + name + '\'' +
+                    '}';
+        }
+    }
+
+    public static class EmployeeDataSerializableFactory implements DataSerializableFactory{
+
+        public static final int FACTORY_ID = 3000;
+
+        @Override
+        public IdentifiedDataSerializable create(int i) {
+            return new IdentifiedDataSerializableEmployee();
+        }
+    }
+
+
+    public static class IdentifiedDataSerializableEmployee implements IdentifiedDataSerializable,EmployeeInterface {
+        protected int id;
+        protected String name;
+
+        public String getName() {
+            return name;
+        }
+
+        public IdentifiedDataSerializableEmployee() {
+        }
+
+        public IdentifiedDataSerializableEmployee(int id) {
+            this.id = id;
+            name = names[random.nextInt(names.length)];
+        }
+
+        @Override
+        public int getFactoryId() {
+            return EmployeeDataSerializableFactory.FACTORY_ID;
+        }
+
+        @Override
+        public int getId() {
+            return 0;
+        }
```

Comment:
I suggest making the code like it should be, no matter if there is a single class the EmployeeDataSerializableFactory needs to deal with. So define a 'id' there, and use it in this method.